pkg/tsdb/stackdriver/stackdriver.go

```diff
@@ -152,43 +152,67 @@ func (e *StackdriverExecutor) buildQueries(tsdbQuery *tsdb.TsdbQuery) ([]*Stackd
 	durationSeconds := int(endTime.Sub(startTime).Seconds())
 
 	for _, query := range tsdbQuery.Queries {
+		migrateLegacyQueryModel(query)
+		q := GrafanaQuery{}
+		model, _ := query.Model.MarshalJSON()
+		if err := json.Unmarshal(model, &q); err != nil {
+			return nil, fmt.Errorf("could not unmarshal StackdriverQuery json: %w", err)
+		}
 		var target string
-
-		metricType := query.Model.Get("metricType").MustString()
-		filterParts := query.Model.Get("filters").MustArray()
-
 		params := url.Values{}
 		params.Add("interval.startTime", startTime.UTC().Format(time.RFC3339))
 		params.Add("interval.endTime", endTime.UTC().Format(time.RFC3339))
-		params.Add("filter", buildFilterString(metricType, filterParts))
-		params.Add("view", query.Model.Get("view").MustString("FULL"))
-		setAggParams(&params, query, durationSeconds)
+
+		sq := &StackdriverQuery{
+			RefID:    query.RefId,
+			GroupBys: []string{},
+		}
+
+		if q.QueryType == metricQueryType {
+			params.Add("filter", buildFilterString(q.MetricQuery.MetricType, q.MetricQuery.Filters))
+			params.Add("view", q.MetricQuery.View)
+			setAggParams(&params, &q.MetricQuery, durationSeconds, query.IntervalMs)
+			sq.GroupBys = append(sq.GroupBys, q.MetricQuery.GroupBys...)
```

Comment:
why group by should be passed as `aggregation.groupByFields` [parameter](https://github.com/grafana/grafana/blob/b8b85b08332f66646668a01e8368de8dae21329f/pkg/tsdb/stackdriver/stackdriver.go#L290) as well as in the `sq.GroupBys`?

Reply:
Good question! That's legacy stuff..I'll remove it! :)

Reply:
...or actually it has to be like that. params is just the URL params for the rest request, whereas sq.GroupBys is a slice that is used to build labels and return meta data.